src/Kupo/App/Database.hs

```diff
@@ -361,6 +368,38 @@ mkDatabase tr longestRollback bracketConnection = Database
                 ] -> yield Input{..}
             (xs :: [SQLData]) -> throwIO (UnexpectedRow addressLike [xs])
 
+    , foldInputsOutputReference = \statusFlag outputRef yield -> ReaderT $ \conn -> do
```

Comment:
Suggestion: rename as `foldInputsByOutputReference` and rename the other as `foldInputsByAddress`